Make.inc

```diff
@@ -497,7 +497,7 @@ JCPPFLAGS_COMMON  := -fasynchronous-unwind-tables
 JCPPFLAGS_CLANG   := $(JCPPFLAGS_COMMON) -mllvm -enable-tail-merge=0
 JCPPFLAGS_GCC     := $(JCPPFLAGS_COMMON) -fno-tree-tail-merge
 
-JCXXFLAGS_COMMON  := -pipe $(fPIC) -fno-rtti -std=c++14
+JCXXFLAGS_COMMON  := -pipe $(fPIC) -fno-rtti -std=c++17
```

Comment:
Does this change the minimum version of compilers required to build julia?

Reply:
If the only new thing is the usage of `std::optional` here, then I think it only raises the baseline for Apple Clang to be Apple Clang 10.0 instead of 9.3. `std::optional` should already be present in GCC 7.1 and Clang 5.0, which are currently the listed minimums.

That said, LLVM says they want GCC 7.4 minimum (and Apple Clang 10), so for a local build of LLVM the GCC minimum will be higher than what we currently say.